client-local/src/main/java/com/cloudera/livy/client/local/LocalClient.java

```diff
@@ -84,6 +85,7 @@
   private final Rpc driverRpc;
   private final ClientProtocol protocol;
   private volatile boolean isAlive;
+  final SparkLauncher launcher = new SparkLauncher();
```

Comment:
This doesn't need to be an instance field, since it's only used in one method and then thrown away.